packages/flutter/lib/src/material/progress_indicator.dart

```diff
@@ -243,6 +243,60 @@ class _LinearProgressIndicatorPainter extends CustomPainter {
 /// The minimum height of the indicator can be specified using [minHeight].
 /// The indicator can be made taller by wrapping the widget with a [SizedBox].
 ///
+/// {@tool dartpad --template=stateful_widget_material_ticker}
+///
+/// This example shows a [LinearProgressIndicator] with a changing value.
+///
+/// ```dart
+/// AnimationController controller;
+/// Animation<double> animation;
+///
+/// @override
+/// void initState() {
+///   controller = AnimationController(
+///     vsync: this,
+///     duration: const Duration(seconds: 5),
+///   );
+///   animation = Tween<double>(
+///     begin: 0,
+///     end: 1,
+///   ).animate(controller)
+///     ..addListener(() {
+///       setState(() {});
+///     });
+///   controller.repeat(reverse: true);
+///   super.initState();
+/// }
```

Comment:
Since `AnimationController` is already an `Animation<double>` that lerps from 0 to 1, the animation isn't necessary here:

```suggestion
///  AnimationController controller;
///
///  @override
///  void initState() {
///    controller = AnimationController(
///      vsync: this,
///      duration: const Duration(seconds: 5),
///    )..addListener(() {
///        setState(() {});
///      });
///    controller.repeat(reverse: true);
///    super.initState();
///  }
```